src/models/mmt_pretrainer.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2021 Google LLC
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     https://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, List
+
+import tensorflow as tf
+from official.nlp.keras_nlp import layers
+
+import layers as mmt_layers
+
+
+class MmtPretrainer(tf.keras.Model):
```

Comment:
As I mentioned on a previous CL, I think "MmtPretraining" sounds more natural than "MmtPretrainer". Consider a name change (even in the comment just below you are already using "pretraining" instead of "pretrainer").

Reply:
Agree! I changed it to `MmtPretrainingModel`.